### PR TITLE
Axi stream monitoring updates

### PR DESCRIPTION
**Pull request by ruck314**
### Description
- adding FrameSize support to AxiStreamMon
-  exposing AXIS_CONFIG_G to register space
- deprecating `AxiStreamMonitoring` device class
- reorg of register space with respect to channels
  - Easier to view when using large number of AXIS lanes

### CODE REVIEW

**Comment by ruck314** [comment]
That would break existing code.  Maybe we can do that during the next "major" release.  I was intending this PR for a future patch release

File: python/surf/axi/_AxiStreamMonAxiL.py [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+            pollInterval = 1,
+        ))
+
+    @staticmethod
+    def convMbps(var):
+        return var.dependencies[0].value() * 8e-6        
+        
+class AxiStreamMonAxiL(pr.Device):
+    def __init__(self,       
+            name        = "AxiStreamMonitoring",
```
Comment (bengineerd) [inline python/surf/axi/_AxiStreamMonAxiL.py]: This should be "AxiStreamMonAxiL".

Also, the Device `name` field gets auto populated with the class name, so in cases where you don't want it different (almost all cases), it's best to just omit it.

Comment (ruck314) [reply python/surf/axi/_AxiStreamMonAxiL.py]: Done. f0212af

**Comment by bengineerd** [comment]
There are only 5 instantiations of it outside of surf itself (and not counting `l2si` which just copied an old surf).

**Comment by ruck314** [comment]
What's about the number times used outside of github/slaclab group or anywhere on gitlab.cern?  I'd prefer not to break existing code unless necessary or part of a major release.

**Comment by bengineerd** [comment]
I'm fine with leaving it until another release. Though now you've got me wondering if it's possible to search for usage at gitlab.cern like we can on github/slaclab.

**Pull request merged**